implementations/rust/ockam/ockam_credential/src/holder.rs

```diff
@@ -0,0 +1,159 @@
+use crate::error::CredentialError;
+use crate::*;
+use bbs::prelude::*;
+use digest::{generic_array::GenericArray, Digest, FixedOutput};
+use ockam_core::lib::*;
+
+/// The label to indicate the secretid attribute in a schema/credential
+pub const SECRET_ID: &'static str = "secretid";
+
+/// Represents a holder of a credential
+#[derive(Debug)]
+pub struct Holder {
+    id: SignatureMessage,
+}
+
+impl Holder {
+    /// Create a new Holder with a new unique id
+    pub fn new() -> Self {
+        Self {
+            id: Prover::new_link_secret(),
+        }
+    }
+
+    /// Accepts a credential offer from an issuer
+    pub fn accept_credential_offer(
+        &self,
+        offer: &CredentialOffer,
+        issuer_pk: &DeterministicPublicKey,
+    ) -> Result<(CredentialRequest, CredentialBlinding), CredentialError> {
+        let nonce = ProofNonce::from(offer.id);
+        let mut i = 0;
+        for (j, att) in offer.schema.attributes.iter().enumerate() {
+            if att.label == SECRET_ID {
+                i = j;
+                break;
+            }
+        }
+
+        let pk = issuer_pk
+            .to_public_key(offer.schema.attributes.len())
+            .map_err(|_| CredentialError::InvalidCredentialSchema)?;
+        let mut messages = BTreeMap::new();
+        messages.insert(i, self.id.clone());
```

Comment:
If there are no schema attributes which have a label of `SECRET_ID` then `i` will be the default initialization of 0. Is that acceptable here?